dashboard/lib/services/user/password_resetter.rb

```diff
@@ -1,10 +1,11 @@
 module Services
   module User
     class PasswordResetter < Services::Base
-      attr_reader :email
+      attr_reader :email, :username
 
-      def initialize(email:)
+      def initialize(email: nil, username: nil)
```

Comment:
I recommend creating two separate service objects: one for resetting the password by email and another for resetting it by username. The new parameters introduce enough additional conditions and complexity to warrant this distinction.